vectordraw/static/js/src/vectordraw.js

```diff
@@ -0,0 +1,633 @@
+            pointSize: 1,
+            pointColor: 'red',
+            width: 4,
+            color: "blue",
+            label: null,
+            labelColor: 'black'
+        };
+        settings.vectors.forEach(function(vector) {
+            _.defaults(vector, default_vector);
+            _.defaults(vector.style, default_vector_style);
+        });
+
+        // Fill in defaults for points.
+        var default_point = {
+            fixed: true,  // Default to true for backwards compatibility.
+            render: true,
+            style: {}
+        };
+        var default_point_style = {
+            size: 1,
+            withLabel: false,
+            color: 'pink',
+            showInfoBox: false
+        };
+        settings.points.forEach(function(point) {
+            _.defaults(point, default_point);
+            _.defaults(point.style, default_point_style);
+            point.style.name = point.name;
+            point.style.fixed = point.fixed;
+            point.style.strokeColor = point.style.color;
+            point.style.fillColor = point.style.color;
+            delete point.style.color;
+        });
+
+        return settings;
+    };
+
+    VectorDraw.prototype.template = _.template([
+        '<div class="jxgboard" style="width:<%= width %>px; height:<%= height %>px;" />',
+        '<div class="menu">',
+        '    <div class="controls">',
+        '        <select>',
+        '        <% vectors.forEach(function(vec, idx) { %>',
+        '            <option value="vector-<%= idx %>"><%= vec.description %></option>',
+        '        <% }) %>',
+        '        <% points.forEach(function(point, idx) { if (!point.fixed) { %>',
+        '            <option value="point-<%= idx %>"><%= point.description %></option>',
+        '        <% }}) %>',
+        '        </select>',
+        '        <button class="add-vector"><%= add_vector_label %></button>',
+        '        <button class="reset">Reset</button>',
+        '        <button class="undo" title="Undo"><span class="fa fa-undo" /></button>',
+        '        <button class="redo" title="redo"><span class="fa fa-repeat" /></button>',
+        '    </div>',
+        '    <% if (show_vector_properties) { %>',
+        '      <div class="vector-properties">',
+        '        <h3><%= vector_properties_label %></h3>',
+        '        <div class="vector-prop-name">',
+        '          name: <span class="value vector-prop-bold">-</span>',
+        '        </div>',
+        '        <div class="vector-prop-length">',
+        '          length: <span class="value">-</span>',
+        '        </div>',
+        '        <div class="vector-prop-angle">',
+        '          angle: <span class="value">-</span>&deg;',
+        '        </div>',
+        '        <div class="vector-prop-slope">',
+        '          slope: <span class="value">-</span>',
+        '        </div>',
+        '      </div>',
+        '    <% } %>',
+        '</div>'
+    ].join('\n'));
```

Comment:
I think we should render this template on the server side now that we can.  It only uses Studio settings and no user state, so it seems like a good fit for being included in the Django template.